src/test_l3_dsFPD.c

```diff
@@ -0,0 +1,716 @@
+    {NULL, -1}
+};
+const static ut_control_keyStringMapping_t dsFrontPanelColorTable[] = {
+    {"dsFPD_COLOR_BLUE", (int32_t)(0x0000FF)},
+    {"dsFPD_COLOR_GREEN", (int32_t)(0x00FF00)},
+    {"dsFPD_COLOR_RED", (int32_t)(0xFF0000)},
+    {"dsFPD_COLOR_YELLOW", (int32_t)(0xFFFFE0)},
+    {"dsFPD_COLOR_ORANGE", (int32_t)(0xFF8C00)},
+    {"dsFPD_COLOR_WHITE", (int32_t)(0xFFFFFF)},
+    {NULL, -1}
+};
+const static ut_control_keyStringMapping_t dsFrontPanelLEDState[] = {
+    {"dsFPD_LED_DEVICE_NONE", (int32_t)dsFPD_LED_DEVICE_NONE},
+    {"dsFPD_LED_DEVICE_ACTIVE", (int32_t)dsFPD_LED_DEVICE_ACTIVE},
+    {"dsFPD_LED_DEVICE_STANDBY", (int32_t)dsFPD_LED_DEVICE_STANDBY},
+    {"dsFPD_LED_DEVICE_WPS_CONNECTING", (int32_t)dsFPD_LED_DEVICE_WPS_CONNECTING},
+    {"dsFPD_LED_DEVICE_WPS_CONNECTED", (int32_t)dsFPD_LED_DEVICE_WPS_CONNECTED},
+    {"dsFPD_LED_DEVICE_WPS_ERROR", (int32_t)dsFPD_LED_DEVICE_WPS_ERROR},
+    {"dsFPD_LED_DEVICE_FACTORY_RESET", (int32_t)dsFPD_LED_DEVICE_FACTORY_RESET},
+    {"dsFPD_LED_DEVICE_USB_UPGRADE", (int32_t)dsFPD_LED_DEVICE_USB_UPGRADE},
+    {"dsFPD_LED_DEVICE_SOFTWARE_DOWNLOAD_ERROR", (int32_t)dsFPD_LED_DEVICE_SOFTWARE_DOWNLOAD_ERROR},
+    {"dsFPD_LED_DEVICE_MAX", (int32_t)dsFPD_LED_DEVICE_MAX},
+    {NULL, -1}
+};
+
+void readAndDiscardRestOfLine(FILE* in)
+{
+    int c;
+    while ( (c = fgetc(in)) != EOF && c != '\n');
+}
+
+/**
+* @brief Initialization of the Device settings - Front Panel Module
+*
+* This test provides a scope to Intialize the Front Panel module.
+
+* **Pre-Conditions:** None@n
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 1 to before running any test.
+*
+*/
+
+void test_1_dsFPD_hal_Init(void)
+{
+   gTestID = 1;
+   dsError_t status = dsERR_NONE;
+
+   UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+   // Step 1: Call dsFPInit()
+   UT_LOG_INFO("Calling dsFPInit()");
+   status = dsFPInit();
+   UT_LOG_INFO("Result dsFPInit(OUT:dsError_t:[%s])",UT_Control_GetMapString(dsFrontPanelErrorCodeTable, status));
+   assert(status == dsERR_NONE);
+
+   UT_LOG_INFO("Out %s\n", __FUNCTION__);
+}
+/**
+* @brief This test provides a scope to set Front panel LED state.
+*
+* This test case provides a scope to  set the State of FP indicator to ON and OFF.
+*
+* **Pre-Conditions:** @n
+* Front Panle Module should be intialized through Test 1 before calling this test.
+*
+* **Dependencies:** None@n
+*
+* **User Interaction:** @n
+* User or Automation tool should select the Test 2 and provide indicator index and State to be set.
+*
+*/
+void test_2_dsFPD_hal_SetFPState(void)
+{
+    gTestID = 2;
+    UT_LOG_INFO("In %s [%02d%03d]\n", __FUNCTION__, gTestGroup, gTestID);
+
+    dsError_t status = dsERR_NONE;
+    int32_t eIndicator;
+    int32_t indicatorType = 0;
+    uint32_t uState = 0;
+    int32_t dsFPDNumberOfIndicators = UT_KVP_PROFILE_GET_UINT32("dsFPD/Number_of_Indicators");
```

Comment:
Please check whether we can make this **"dsFPDNumberOfIndicators"** as global. Since, you are using it in most of the API's.

Reply:
moved to separate fucntion